Tasks/Test1/Test1.Tests/CheckSumUtilsTest.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2024
+//
+// Use of this source code is governed by an MIT license
+// that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+namespace CheckSum.Tests;
+
+public static class CheckSumUtilsTest
+{
+    private const string TestFilesPath = "TestFiles/";
+
+    public static IEnumerable<TestCaseData> TestCases()
+    {
+        foreach (var testFile in Directory.GetFileSystemEntries(TestFilesPath))
+        {
+            yield return new TestCaseData(testFile);
+        }
+    }

```

Comment:
```suggestion
    public static IEnumerable<TestCaseData> TestCases() 
        => Directory.GetFileSystemEntries(TestFilesPath).Select(e => new TestCaseData(e));
```
Если проходите по коллекции, чтобы сделать yield, это сразу знак, что делаете что-то не так, коллекция ведь и так коллекция, а yield нужен, чтобы по набору значений коллекцию сделать